x-pack/plugins/observability_solution/observability/server/lib/annotations/create_annotations_client.ts

```diff
@@ -79,7 +74,43 @@ export function createAnnotationsClient(params: {
         const body = await unwrapEsResponse(
           esClient.index(
             {
-              index,
+              index: index + `-v${ANNOTATION_RESOURCES_VERSION}`,
```

Comment:
Does this somewhat break the expectation that users can use their own index name, if we're appending things onto it?